app/controllers/appointments_controller.rb

```diff
@@ -1,11 +1,41 @@
 class AppointmentsController < ApplicationController
   def show
-    @appointments = Appointments.new(['Thursday Afternoon (11th October)'])
+    @form = AppointmentForm.new
+    @appointments = available_appointments
   end
 
   def submit
-    redirect_to confirmation_path(params[:work_order_reference])
+    @form = AppointmentForm.new(appointment_form_params)
+
+    if @form.invalid?
+      @appointments = available_appointments
+      return render :show
+    end
+
+    # TODO: Book appointment
+
+    redirect_to confirmation_path(params[:repair_request_reference])
+  end
+
+  private
+
+  def appointment_form_params
+    params.require(:appointment_form).permit(:appointment_id)
+  end
+
+  def work_order_reference
```

Comment:
The original idea was that `Appointments` would be a facade exposing `form` and `slots_for_collection` and providing a place for these methods to live so that we don't have to add them in the controller.

This approach works for now but would it be more flexible to move these methods into an object?

Reply:
I'm going to leave this in the controller for now I think. If it gets any more complex, I'll revisit and extract this into smaller objects.